samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java

```diff
@@ -594,7 +593,13 @@ public void deleteMessages(Map<SystemStreamPartition, String> offsets) {
           .collect(Collectors.toMap(entry ->
               new TopicPartition(entry.getKey().getStream(), entry.getKey().getPartition().getPartitionId()),
               entry -> RecordsToDelete.beforeOffset(Long.parseLong(entry.getValue()) + 1)));
-      adminClient.deleteRecords(recordsToDelete);
+
+      adminClient.deleteRecords(recordsToDelete).all().whenComplete((ignored, exception) -> {
```

Comment:
Just to double check, `whenComplete` is async, right?

Reply:
Yes